### PR TITLE
[Merged by Bors] - feat(Probability): add `indepFun_of_identDistrib_pair `

### PR DESCRIPTION
**Pull request by pitmonticone**
This PR adds the lemma claiming that if `X` and `Y` are independent and `(X, Y)` and `(X', Y')` are identically distributed, then `X'` and `Y'` are independent.

See the associated [Zulip topic](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/.60IndepFun.60.20of.20.60IdentDistrib.60.20pairs).

### CODE REVIEW

File: Mathlib/Probability/IdentDistrib.lean [hunk 1]
```diff
@@ -366,4 +366,19 @@ theorem Memℒp.uniformIntegrable_of_identDistrib {ι : Type*} {f : ι → α 
 
 end UniformIntegrable
 
+/-- If `X` and `Y` are independent and `(X, Y)` and `(X', Y')` are identically distributed,
+then `X'` and `Y'` are independent. -/
+lemma indepFun_of_identDistrib_pair
+    {μ : Measure γ} {μ' : Measure δ} [IsFiniteMeasure μ] [IsFiniteMeasure μ']
+    {X : γ → α} {X' : δ → α} {Y : γ → β} {Y' : δ → β} (hX : AEMeasurable X μ)
+    (hX' : AEMeasurable X' μ') (hY : AEMeasurable Y μ) (hY' : AEMeasurable Y' μ')
```
Comment (JasonKYi) [inline Mathlib/Probability/IdentDistrib.lean:374]: The AEMeasurable assumptions are unnecessary as you can get them from `IdentDistrib.aemeasurable_fst`, e.g.
```lean
example {μ : Measure γ} {μ' : Measure δ} [IsFiniteMeasure μ] [IsFiniteMeasure μ']
    {X : γ → α} {X' : δ → α} {Y : γ → β} {Y' : δ → β}
    (h_ident : IdentDistrib (fun ω ↦ (X ω, Y ω)) (fun ω ↦ (X' ω, Y' ω)) μ μ') :
    AEMeasurable X μ := by
    rw [(by ext; simp : X = Prod.fst ∘ (fun ω ↦ (X ω, Y ω)))]
    refine AEMeasurable.comp_aemeasurable measurable_fst.aemeasurable h_ident.aemeasurable_fst
```

Comment (pitmonticone) [reply Mathlib/Probability/IdentDistrib.lean]: Done. Thank you very much for your review.

File: Mathlib/Probability/IdentDistrib.lean [hunk 2]
```diff
@@ -366,4 +366,29 @@ theorem Memℒp.uniformIntegrable_of_identDistrib {ι : Type*} {f : ι → α 
...
+    rw [(by ext; simp : X' = Prod.fst ∘ (fun ω ↦ (X' ω, Y' ω)))]
+    exact measurable_fst.aemeasurable.comp_aemeasurable h_ident.aemeasurable_snd
+  have hY' : AEMeasurable Y' μ' := by
+    rw [(by ext; simp : Y' = Prod.snd ∘ (fun ω ↦ (X' ω, Y' ω)))]
+    exact measurable_snd.aemeasurable.comp_aemeasurable h_ident.aemeasurable_snd
+  apply (indepFun_iff_map_prod_eq_prod_map_map hX' hY').mpr
+  have iX : IdentDistrib X X' μ μ' := h_ident.comp measurable_fst
+  have iY : IdentDistrib Y Y' μ μ' := h_ident.comp measurable_snd
+  rw [← h_ident.map_eq, ← iX.map_eq, ← iY.map_eq]
+  exact indepFun_iff_map_prod_eq_prod_map_map hX hY |>.mp h_indep
```
Comment (JasonKYi) [inline Mathlib/Probability/IdentDistrib.lean:392]: Some golf:
```suggestion
  rw [indepFun_iff_map_prod_eq_prod_map_map _ _, ← h_ident.map_eq,
    (indepFun_iff_map_prod_eq_prod_map_map _ _).1 h_indep]
  . exact congr (congrArg Measure.prod <| (h_ident.comp measurable_fst).map_eq)
      (h_ident.comp measurable_snd).map_eq
  . exact measurable_fst.aemeasurable.comp_aemeasurable h_ident.aemeasurable_fst
  . exact measurable_snd.aemeasurable.comp_aemeasurable h_ident.aemeasurable_fst
  . exact measurable_fst.aemeasurable.comp_aemeasurable h_ident.aemeasurable_snd
  . exact measurable_snd.aemeasurable.comp_aemeasurable h_ident.aemeasurable_snd
```

**Comment by JasonKYi** [comment]
Thanks!
maintainer merge

**Pull request closed without merging**